src/bp/ui-admin/src/Pages/Login.js

```diff
@@ -82,26 +104,45 @@ export default class Login extends Component {
     )
   }
 
+  renderExternal() {
+    return (
+      <Fragment>
+        <CardTitle>Botpress Admin Panel</CardTitle>
+        <CardText>Login</CardText>
+        {this.state.error && <Alert color="danger">{this.state.error}</Alert>}
+        <p>
+          <Button onClick={this.redirectToExternalAuthProvider}>Sign in with SSO</Button>
+        </p>
+      </Fragment>
+    )
+  }
+
   render() {
     if (this.props.auth.isAuthenticated()) {
       return <Redirect to="/" />
     }
 
-    if (this.state.isFirstTimeUse) {
+    if (this.state.isFirstTimeUse && this.state.authStrategy === 'basic') {
       return <Redirect to="/register" />
     }
 
     if (this.state.passwordExpired) {
       return <Redirect to={{ pathname: '/ChangePassword', state: this.state }} />
     }
 
+    if (this.state.isLoading) {
+      return null
+    }
+
     return (
       <div className="centered-container">
         <div className="middle">
           <div className="inner">
             <img className="logo" src={logo} alt="loading" />
             <Card body>
-              <CardBody className="login-box">{this.renderForm()}</CardBody>
+              <CardBody className="login-box">
+                {this.state.authStrategy === 'saml' ? this.renderExternal() : this.renderForm()}
```

Comment:
what if it's ldap ? don't you render external ?

Reply:
Nop, since you need the login form as well